torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -105,6 +105,10 @@ class TORCH_API Reducer {
   // Cannot combine with the call of `register_comm_hook`.
   void register_builtin_comm_hook(c10d::BuiltinCommHookType comm_hook_type);
 
+  // Runs allreduce or installed communication hook given GradBucket instance.
+  c10::intrusive_ptr<c10::ivalue::Future> run_reduction_hook(
```

Comment:
Shall we define it as a private method instead? I expect this is just an internal helper function right?

Reply:
I am planning to call this in `distributed.py` so it at least needs to be pybinded as a public method.

Reply:
Does this mean that you want to be able to kick off a separate allreduce at somewhere else?

Reply:
Yeah, when running with uneven inputs context manager, some of the processes in the entire world can already be finished training. To avoid hangs we then simulate an allreduce with zero tensors to match the allreduce of the currently active trainers.